packages/server/lib/controllers/sync/deploy/postDeployInternal.ts

```diff
@@ -0,0 +1,133 @@
+import { z } from 'zod';
+import { zodErrorToHTTP } from '@nangohq/utils';
+import type { PostDeployInternal } from '@nangohq/types';
+import { asyncWrapper } from '../../../utils/asyncWrapper.js';
+import { deploy, errorManager, getAndReconcileDifferences, environmentService, configService, connectionService } from '@nangohq/shared';
+import { getOrchestrator } from '../../../utils/utils.js';
+import { logContextGetter } from '@nangohq/logs';
+import { flowConfigs, jsonSchema, postConnectionScriptsByProvider } from './postConfirmation.js';
+
+const orchestrator = getOrchestrator();
+
+const validation = z
+    .object({
+        flowConfigs: flowConfigs,
+        postConnectionScriptsByProvider: postConnectionScriptsByProvider,
+        jsonSchema: jsonSchema.optional(),
+        nangoYamlBody: z.string(),
+        reconcile: z.boolean(),
+        debug: z.boolean(),
+        singleDeployMode: z.boolean().optional().default(false)
+    })
+    .strict();
+
+const queryStringValidation = z
+    .object({
+        customEnvironment: z.string().min(1)
+    })
+    .strict();
+
+export const postDeployInternal = asyncWrapper<PostDeployInternal>(async (req, res) => {
+    const val = validation.safeParse(req.body);
+    if (!val.success) {
+        res.status(400).send({ error: { code: 'invalid_body', errors: zodErrorToHTTP(val.error) } });
+        return;
+    }
+
+    const queryParamValues = queryStringValidation.safeParse(req.query);
+
+    if (!queryParamValues.success) {
+        res.status(400).send({
+            error: { code: 'invalid_query_params', errors: zodErrorToHTTP(queryParamValues.error) }
+        });
+        return;
+    }
+
+    const body: PostDeployInternal['Body'] = val.data;
+
+    const { account } = res.locals;
+
+    if (account.uuid !== process.env['NANGO_SHARED_DEV_ACCOUNT_UUID']) {
+        res.status(403).send({ error: { code: 'forbidden', message: 'This endpoint is only available for Nango internal use' } });
+        return;
+    }
+
+    const environmentName = queryParamValues.data.customEnvironment;
+
+    let environment = await environmentService.getByEnvironmentName(account.id, environmentName);
+
+    if (!environment) {
+        environment = await environmentService.createEnvironment(account.id, environmentName);
+
+        if (!environment) {
+            res.status(500).send({
+                error: { code: 'environment_creation_error', message: 'There was an error creating the environment, please try again' }
+            });
+            return;
+        }
+
+        // since we're making a new environment, we want to make sure the config creds and
+        // connections are copied from the dev environment
+        // so that the syncs and actions can be run right away
```

Comment:
It feels unnecessary, we have about 60 connections in total that could have been moved manually